### DISCUSSION TITLE
Using systemd drop-in file for salt-minion.service override

### DISCUSSION THREAD

**Discussion by petiepooo** [Security-Onion-Solutions/securityonion#8380] [Ideas] [answered]
Replacing the upstream packager's service file can lead to subtle bugs if an upstream change goes unnoticed.  For example, at 
https://github.com/Security-Onion-Solutions/securityonion/blob/484aa7b2075eb7abb8c53cf021a90d1bd6bf8211/salt/salt/service/salt-minion.service.jinja#L12 it appears the upstream _salt-minion.service_ file is replaced in order to add a 30 second startup delay.

This is probably better accomplished by creating a drop-in file at say _/etc/systemd/system/salt-minion.service.d/start-delay.conf_ containing just the results of jinja processing the following:
> [Service]
ExecStartPre=/bin/sleep {{ salt['pillar.get']('salt:minion:service_start_delay', service_start_delay) }}

and leaving the upstream packager's service file in _/lib/systemd/system_ active.

An example of a subtle bug: SaltStack changed their base URL from _saltstack.com_ to _saltproject.io_, but the documentation reference has not been updated in your copy of their service file.  This results in the (harmless now but maybe not so in the future) salt output during so-setup of

> [INFO    ] Executing state file.managed for [/etc/systemd/system/multi-user.target.wants/salt-minion.service]
[INFO    ] Fetching file from saltenv 'base', ** done ** 'salt/service/salt-minion.service.jinja'
[INFO    ] File changed:
> \---
> +++ 
@@ -1,6 +1,6 @@
 [Unit]
 Description=The Salt Minion
-Documentation=man:salt-minion(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltproject.io/en/latest/contents.html
+Documentation=man:salt-minion(1) file:///usr/share/doc/salt/html/contents.html https://docs.saltstack.com/en/latest/contents.html
 After=network.target salt-master.service
>  
> [Service]
@@ -9,6 +9,7 @@
 NotifyAccess=all
 LimitNOFILE=8192
 ExecStart=/usr/bin/salt-minion
+ExecStartPre=/bin/sleep 30
> 
> [Install]
-WantedBy=multi-user.target
+WantedBy=multi-user.target
[INFO    ] Completed state [/etc/systemd/system/multi-user.target.wants/salt-minion.service] at time 15:01:42.444921 (duration_in_ms=17.01)

For more details on using systemd's drop-in unit files, search for the keyword 'drop-in' in the manpage at https://www.freedesktop.org/software/systemd/man/systemd.unit.html

**Comment by dougburks** [comment, marked as answer]
Created issue https://github.com/Security-Onion-Solutions/securityonion/issues/8441
